### PR TITLE
Adapt to conduit 2.3

### PR DESCRIPTION
**Pull request by samoht**
Adapting to https://github.com/mirage/ocaml-conduit/pull/376 and https://github.com/mirage/ocaml-cohttp/pull/742

### CODE REVIEW

File: lib/mirage_impl_http.ml [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
 
+let cohttp_server conduit = cohttp_server $ conduit
+
+let cohttp_client = impl @@ object
+    inherit base_configurable
+    method ty = pclock @-> resolver @-> conduit @-> http_client
+    method name = "http_client"
+    method module_name = "Cohttp_mirage.Client.Make"
+    method! packages =
+      Mirage_key.pure [ package ~min:"3.0.0" ~max:"4.0.0" "cohttp-mirage" ]
```
Comment (dinosaure) [inline lib/mirage_impl_http.ml:33]: May be this is wrong due to the new release of `cohttp`?

File: lib/mirage_impl_http.mli [hunk 2]
```diff
@@ -1,9 +1,19 @@
+open Functoria
+
 type http
 
-val http : http Functoria.typ
+val http : http typ
+
+val cohttp_server : Mirage_impl_conduit.conduit impl -> http impl
+
+val httpaf_server : Mirage_impl_conduit.conduit impl -> http impl
```
Comment (dinosaure) [inline lib/mirage_impl_http.mli:9]: I'm not sure that it's relevant to continue to support it when we know that `Httpaf_mirage` has some bugs (/cc @TheLortex) and when `paf` exists and it is used. At least, can you delete the `httpaf` support? I will come back with the `paf` support.

**Comment by dinosaure** [comment]
I'm ok to merge but small details remains. I would like to have the opinion/CI result from @TheLortex too.

**Comment by hannesm** [comment]
I'm very much in favor to get this merged and released, especially with the situation now: mirage 3.10 emits `conduit >= 2.2.0 & < 3.0.0` dependencies, which install conduit 2.3.0 (and cohttp 4.0.0), and compilation fails (try to compile e.g. https://github.com/roburio/dns-letsencrypt-secondary with an up-to-date opam-repository).

In the future, we should bump major revisions (i.e. conduit-mirage 2.3.0 should be 3.0.0 (or 4.0.0 in case the unavailable 3.0.0 is supposed to stay there) when the API changes.

**Comment by dinosaure** [comment]
> I'm very much in favor to get this merged and released, especially with the situation now: mirage 3.10 emits conduit >= 2.2.0 & < 3.0.0 dependencies, which install conduit 2.3.0 (and cohttp 4.0.0), and compilation fails (try to compile e.g. https://github.com/roburio/dns-letsencrypt-secondary with an up-to-date opam-repository).

Me too 👍 I will try to update the PR from my comments.

**Comment by dinosaure** [comment]
I think it's ready to merge, it's ok to merge as is? (@hannesm, @samoht) and cut a release?

**Comment by hannesm** [comment]
I have not reviewed this patch, nor followed conduit / cohttp API changes, nor tested this patch to work (but I trust you & Thomas). Looking at the mirage.mli changes in this PR this looks fine (and very backwards-compatible). A merge and release would be great.

**Comment by dinosaure** [comment]
I tested locally and it's work, let's merge.

**Pull request merged**